src/time.c

```diff
@@ -46,6 +46,44 @@ rcutils_time_point_value_as_nanoseconds_string(
   return RCUTILS_RET_OK;
 }
 
+rcutils_ret_t
+rcutils_time_point_value_as_date_string(
+  const rcutils_time_point_value_t * time_point,
+  char * str,
+  size_t str_size)
+{
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(time_point, RCUTILS_RET_INVALID_ARGUMENT);
+  RCUTILS_CHECK_ARGUMENT_FOR_NULL(str, RCUTILS_RET_INVALID_ARGUMENT);
+  if (0 == str_size) {
+    return RCUTILS_RET_OK;
+  }
+  // best to abs it to avoid issues with negative values in C89, see:
+  //   https://stackoverflow.com/a/3604984/671658
+  uint64_t abs_time_point = (uint64_t)llabs(*time_point);
+  // break into two parts to avoid floating point error
+  uint64_t seconds = abs_time_point / (1000u * 1000u * 1000u);
+  uint64_t nanoseconds = abs_time_point % (1000u * 1000u * 1000u);
+  char str2[21];   // Make sure the buffer is large enough to hold the largest possible uint64_t
+
+  if (rcutils_snprintf(str2, sizeof(str2), "%" PRIu64, nanoseconds)) {
+    RCUTILS_SET_ERROR_MSG("failed to format time point into string as iso8601_date");
+    return RCUTILS_RET_ERROR;
+  }
+
+  time_t now_t = (time_t)(seconds);
+  struct tm ptm = {.tm_year = 2022, .tm_mday = 1};
+  localtime_r(&now_t, &ptm);
+
+  strftime(str, 32, "%Y-%m-%d %H:%M:%S", &ptm);
+  int position = 19;
```

Comment:
```suggestion
  static const int position = 19;
```

This could also use a comment explaining why "19", and/or renaming the variable to be more clear.